subsys/bluetooth/mesh/dfu_cli.c

```diff
@@ -965,11 +965,6 @@ static int dfu_cli_init(const struct bt_mesh_model *mod)
 {
 	struct bt_mesh_dfu_cli *cli = mod->rt->user_data;
 
-	if (mod->rt->elem_idx != 0) {
-		LOG_ERR("DFU update client must be instantiated on first elem");
-		return -EINVAL;
-	}
-
 	cli->mod = mod;
 
 	if (IS_ENABLED(CONFIG_BT_MESH_MODEL_EXTENSIONS)) {
```

Comment:
Here I only remove what is wrong, forcing dfu client to first element. The issue you mentioned is correct however is not in this commits scope.

also the helper macros for dfu and dfd currently covers most part, https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/bluetooth/mesh/dfu_cli.h#L48
but eventually it'd be nice have have check here when helpers are not used.

Reply:
Checking with zero element index implicitly checked that blob and dfu clients are on the same element. Since dfu client extends blob client, they must be on the same element. This is the mandatory statement is not checked anyhow. 
```
7.1.1 Description
The Firmware Update Client is a main model that extends the BLOB Transfer Client model
```

Reply:
Yes, Zephyr provides helper macros to avoid misconfiguration. However, there is no guarantee the macro was used.

Reply:
I agree with you from the very beginning however currently this situation is not a bug, and adding check to models being in the same element depending on device role is kind of improvement and it is not scope of this PR. Also because this check should be added to other dfu/dfd/mbt models. Maybe it would be clearer and helpful if helpers named dfu_target, dfu_initiator, dfu_distributor and dfu_standalone_updater, because IMO currently helpers are not clear in purpose, dfd_srv, so what?

In brief that sounds like an imrovement more than a bug.

Reply:
What we need is to copy this 7 lines of code really:
https://github.com/zephyrproject-rtos/zephyr/blob/3364a35f05475a947dfd0bfc4f7b8984504caee4/subsys/bluetooth/mesh/priv_beacon_srv.c#L198-L204

Reply:
If we don't do this, absolutely anything can happen at the `bt_mesh_model_extend` call.

Reply:
then it is need to be done for every models which does not have this check?

Reply:
I think we spend more time arguing here than adding this check.

Reply:
applied the same check logic also for dfd_srv and dfu_srv to check required models are presented in the same element.